pkg/controller/clone/planner.go

```diff
@@ -431,6 +502,141 @@ func (p *Planner) planHostAssistedFromPVC(ctx context.Context, args *PlanArgs) (
 	return []Phase{hcp, rp}, nil
 }
 
+func (p *Planner) planHostAssistedFromSnapshot(ctx context.Context, args *PlanArgs) ([]Phase, error) {
+	sourceSnapshot := &snapshotv1.VolumeSnapshot{}
+	exists, err := getResource(ctx, p.Client, args.DataSource.Namespace, args.DataSource.Spec.Source.Name, sourceSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("source claim does not exist")
+	}
+
+	vsc, err := GetCompatibleVolumeSnapshotClass(ctx, p.Client, args.TargetClaim)
+	if err != nil {
+		return nil, err
+	}
+	if vsc == nil {
+		return nil, fmt.Errorf("no compatible volumesnapshotclass")
+	}
+
+	sourceClaimForDumbClone, err := createTempSourceClaim(ctx, args.DataSource.Namespace, args.TargetClaim, sourceSnapshot, p.Client)
```

Comment:
This claim should be the same size as the `restoreSize` of the snapshot.  We don't want prep phase to expand it to the target size.

Reply:
We still want to prep `sourceClaimForDumbClone`, right? Is it enough to use the restoreSize when creating the claim?

Reply:
yes prep is for force bind only if required